wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/WorkspaceManager.java

```diff
@@ -264,6 +271,15 @@ public WorkspaceImpl updateWorkspace(String id, Workspace update)
    */
   @Traced
   public void removeWorkspace(String workspaceId) throws ConflictException, ServerException {
+    TracingTags.WORKSPACE_ID.set(workspaceId);
+    TracingTags.USER_ID.set(() -> EnvironmentContext.getCurrent().getSubject().getUserId());
+    try {
+      TracingTags.STACK_ID.set(
```

Comment:
This```STACK_ID`` code looks overweight. Maybe we can remove it at all? @sleshchenko @mshaposhnik @metlos wdyt?

Reply:
The code could be rewritten here in more elegant way. More important question - can information about stack id of removed workspace be useful? If yes - then we should set it.
@garagatyi @ibuziuk Guys, do you think it can be important for OSIO installation?

Reply:
Aren't we going to remove stacks in Che 7? Not sure it is that useful to add stack ID

Reply:
@garagatyi I can simplify retrieval of stackId by having removeWorkspace method of dao to return the entity of workspace (so no extra call of dao method will be needed)

Reply:
simplified